src/rules/effects/no-multiple-actions-in-effects.ts

```diff
@@ -21,6 +21,7 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
       category: 'Best Practices',
       description: 'An `Effect` should not return multiple actions.',
       recommended: 'warn',
+      requiresTypeChecking: true,
```

Comment:
While I understand the point of `avoid-cyclic effects` being a rule that requires type checking, this one I don't understand why it needs to be either. 
I mean, users may have a [performance loss](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md#performance) to report a rare case like this:

```ts
export class Effects {
  test$ = createEffect(() =>
    this.actions$.pipe(concatMap(() => {
      const actions: Action[] = [];
      return actions;
    }))
  )
}
```

Unless I'm missing something, IMHO it shouldn't require type-checking as the other rules we already have.

Reply:
It's indeed for that case. 
I've added this because I saw the following code in one of our projects:
The only way I could report this was by using type checking.

```ts
export class Effects {
  test$ = createEffect(() =>
    this.actions$.pipe(concatMap(() => {
      const actions: Action[] = [];
      if(condition) actions.push(action1);
      if(condition2) actions.push(action2);
      if(condition3) actions.push(action3);
      return actions;
    }))
  )
}
```